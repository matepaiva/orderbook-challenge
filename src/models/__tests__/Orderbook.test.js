const { Document } = require('camo');
const { Orderbook } = require('../index');

jest.mock('../modifiers', () => ({
    withTimestamps: (val) => val,
}));

jest.mock('camo', () => ({
    Document: class MockDocument { },
}));


describe('Orderbook', () => {
    describe('.constructor()', () => {
       it('should define fields', () => {
           const orderbook = new Orderbook();

           expect(orderbook).toMatchSnapshot();
           expect(orderbook).toBeInstanceOf(Document);
       });
    });

    describe('.collectionName()', () => {
        it('should return the collection name', () => {
            expect(Orderbook.collectionName()).toBe('orderbooks');
        });
    });
});
