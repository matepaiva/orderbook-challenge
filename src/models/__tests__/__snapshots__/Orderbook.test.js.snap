// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orderbook .constructor() should define fields 1`] = `
Orderbook {
  "age": Object {
    "type": [Function],
  },
  "name": Object {
    "required": true,
    "type": [Function],
    "unique": true,
  },
}
`;
